src/generic_core/metrics.ts

```diff
@@ -157,7 +157,23 @@ function sumMetrics(mets :NetMetrics) :number{
   }
 }
 
-export class Metrics {
+export interface Metrics {
+  increment(name :string) : void;
+  userCount(network: string, userName :string, friendCount:number) : void;
+  getReport(natInfo:uproxy_core_api.NetworkInfo) : Promise<Object>;
+  reset() : void;
+}
+
+export class NoOpMetrics implements Metrics {
+  public increment(name :string) {}
+  public userCount(network: string, userName :string, friendCount:number) {}
+  public getReport(natInfo:uproxy_core_api.NetworkInfo) {
+    return new Promise((F, R) => {});  // Never provide a report.
```

Comment:
I expanded the comment here to explain what's going on
```typescript
// To avoid contaminating the production metrics with empty reports, we
// return a promise that never resolves at all.
```
Unless we make some other changes, using `Promise.resolve` here would have a different effect, posting empty stats reports to the reporting server.